src/main/java/org/gridsuite/sensitivityanalysis/server/service/nonevacuatedenergy/NonEvacuatedEnergyInputBuilderService.java

```diff
@@ -0,0 +1,514 @@
+                        .withValue("limitName", branches.getLimitNameN())
+                        .withValue("id", branch.getId())
+                        .build());
+                    throw new PowsyblException("Temporary limit '" + branches.getLimitNameN() + "' not found for branch '" + branch.getId() + "' !!");
+                }
+            }
+
+            if (StringUtils.isNotEmpty(branches.getLimitNameNm1())) {  // here we consider the temporary limit
+                // temporary limits on side 1
+                boolean factors1Generated = genFactorForTemporaryLimit(branch, Branch.Side.ONE, currentLimits1, branches.getLimitNameNm1(), false, variableIds, variableSet, sensitivityVariableType, branches, monitoredBranchThreshold, contingencies, result, reporter);
+
+                // temporary limits on side 2
+                boolean factors2Generated = genFactorForTemporaryLimit(branch, Branch.Side.TWO, currentLimits2, branches.getLimitNameNm1(), false, variableIds, variableSet, sensitivityVariableType, branches, monitoredBranchThreshold, contingencies, result, reporter);
+
+                if (!factors1Generated && !factors2Generated) {
+                    reporter.report(Report.builder()
+                        .withKey("monitoredBranchTemporaryLimitNotFound")
+                        .withDefaultMessage("Temporary limit ${limitName} not found for the monitored branch ${id}")
+                        .withSeverity(TypedValue.ERROR_SEVERITY)
+                        .withValue("limitName", branches.getLimitNameNm1())
+                        .withValue("id", branch.getId())
+                        .build());
+                    throw new PowsyblException("Temporary limit '" + branches.getLimitNameNm1() + "' not found for branch '" + branch.getId() + "' !!");
+                }
+            }
+
+            if (!result.isEmpty()) {
+                context.getNonEvacuatedEnergyInputs().getBranchesThresholds().put(monitoredEquipment.getId(), monitoredBranchThreshold);
+            }
+        });
+
+        return result;
+    }
+
+    private void buildSensitivityFactorsAndVariableSets(NonEvacuatedEnergyRunContext context,
+                                                        Network network,
+                                                        Reporter reporter) {
+        List<NonEvacuatedEnergyMonitoredBranches> monitoredBranches = context.getNonEvacuatedEnergyInputData().getNonEvacuatedEnergyMonitoredBranches();
+        List<NonEvacuatedEnergyGeneratorLimitByType> generatorsLimitByType = context.getNonEvacuatedEnergyInputData().getNonEvacuatedEnergyGeneratorsLimit().getGenerators();
+
+        // build inputs for the sensitivities in MW per generation kind (similar to sensitivity analysis computation with injections set)
+        generatorsLimitByType.stream()
+            .filter(NonEvacuatedEnergyGeneratorLimitByType::isActivated) // we keep only activated generators limits
```

Comment:
Should we warn the user in the case where he inputs filters that are not of the correct selected type?
For example, he selects "solar" and he inputs equipments with hydro type in the filter.

Reply:
Done